refiner-app/src/store.rs

```diff
@@ -99,16 +99,21 @@ async fn save_last_block_height<P: AsRef<std::path::Path>>(storage_path: P, bloc
     if !path.exists() {
         tokio::fs::create_dir_all(path).await.unwrap();
     }
-    let path = path.join(STORE_INFO_FILE);
+    let file_path = path.join(STORE_INFO_FILE);
 
-    let file = tokio::fs::File::create(path).await.unwrap();
+    // Write the data to a height-specific file to avoid clearing the main file
+    let temp_path = path.join(format!(".{block_height}"));
+    let temp_file = tokio::fs::File::create(&temp_path).await.unwrap();
 
     {
-        let mut writer = tokio::io::BufWriter::new(file);
+        let mut writer = tokio::io::BufWriter::new(temp_file);
         let data = serde_json::to_string(&block_height).unwrap();
```

Comment:
It seems that serde usage here is overkill 😄 
```suggestion
        writer.write_all(data.to_string().as_bytes()).await.unwrap();
```

Reply:
```rust
async fn save_last_block_height<P: AsRef<std::path::Path>>(storage_path: P, block_height: u64) {
    let path = storage_path.as_ref();
    if !path.exists() {
        tokio::fs::create_dir_all(path).await.unwrap();
    }
    let file_path = path.join(STORE_INFO_FILE);

    tokio::fs::write(file_path, block_height.to_string())
        .await
        .unwrap();
    tracing::trace!("Last block height {} saved.", block_height);
}
```
Shorter version.

Reply:
I don't think it works atomically to simply call `tokio::fs::write`. It could still be possible for an unexpected shutdown to leave `STORE_INFO_FILE` in a corrupted (empty) state. The point of writing to another file and then using `rename` is that the initial write does not change `STORE_INFO_FILE`, so an interrupt at that point leaves the file in tact. And `rename` itself is safe because it uses OS functions under the hood which are atomic, regardless of what happens to the running application.

I agree the serde call is not needed though.

Reply:
And trivial test could be added:
```rust
#[tokio::test]
async fn test_save_last_block_height() {
    let tmp_dir = std::env::temp_dir();
    save_last_block_height(&tmp_dir, 11111).await;
    let block_height = load_last_block_height(tmp_dir).await;

    assert_eq!(block_height, Some(11111))
}
```

Reply:
Good call. Done in https://github.com/aurora-is-near/borealis-engine-lib/pull/34/commits/d9d83d4bae498ebbf1b454df4207e30e2c20517b